executor/habitat.go

```diff
@@ -1,40 +1,94 @@
 package executor
 
 import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/screwdriver-cd/sd-cmd/config"
+	"github.com/screwdriver-cd/sd-cmd/screwdriver/store"
 	"github.com/screwdriver-cd/sd-cmd/util"
 )
 
 var habPath = "/hab/bin/hab"
 
 // Habitat is the Habitat Executor struct
 type Habitat struct {
-	Args []string
-	Spec *util.Habitat
+	Args  []string
+	Spec  *util.CommandSpec
+	Store store.Store
 }
 
 // NewHabitat returns the Habitat struct
 func NewHabitat(spec *util.CommandSpec, args []string) (habitat *Habitat, err error) {
+	storeapi := store.New(config.SDStoreURL, spec, config.SDToken)
+
 	habitat = &Habitat{
-		Args: args,
-		Spec: &util.Habitat{
-			Mode:    spec.Habitat.Mode,
-			Package: spec.Habitat.Package,
-			Command: spec.Habitat.Command,
-		},
+		Args:  args,
+		Spec:  spec,
+		Store: storeapi,
 	}
 	return habitat, nil
 }
 
+func (h *Habitat) getPkgDirPath() string {
+	return filepath.Join(config.BaseCommandPath, h.Spec.Namespace, h.Spec.Name, h.Spec.Version)
```

Comment:
Is `h.Spec.Hbitat...` correct?

Reply:
That's not layer of `h.Spec.Habitat`. Please ignore this.

Reply:
🙆